app/Models/TrainCheckin.php

```diff
@@ -128,7 +127,7 @@ public function getDurationAttribute(): int {
     }
 
     public function getSpeedAttribute(): float {
-        return $this->duration == 0 ? 0 : $this->distance / ($this->duration / 60);
+        return $this->duration == 0 ? 0 : ($this->distance / 1000) / ($this->duration / 60);
```

Comment:
🤓
```suggestion
        return $this->duration == 0 ? 0 : ($this->distance) / ($this->duration * 60) * 3.6;
```
(have not tested it though)